tech-far-hub/src/types.ts

```diff
@@ -51,10 +53,10 @@ export type MinimalPageQuery = {
   readonly siblings: { readonly nodes: ReadonlyArray<{ readonly frontmatter: IMinimalFrontmatter | null }> };
 };
 
-export type TOCEnhancedQueryPageProps<T extends MinimalPageQuery> = PageProps<
-  Omit<T, "currentPage"> & (DeepPick<T, "currentPage.!tableOfContents"> & TableOfContents),
```

Comment:
Did the original `DeepPick<T, "currentPage.!tableOfContents">` give a TS warning?

Reply:
It did, and this does too. But now at least it's less hard to read.